### PR TITLE
Add test to replicate duplicate port hints issue

### PR DESCRIPTION
**Pull request by abhijitxy**
- [x] Added a test to replicate the issue of duplicate port hints by creating a `bug` with `footprint="soic8"`.

### CODE REVIEW

**Comment by abhijitxy** [comment]
@seveibar review it

**Review by seveibar — CHANGES_REQUESTED** [review]
Oh i would make this test fail (duplicates are a bug) then fix the duplicate port hints issue. Wasnt sure if you were waiting on a review, if we merged as is we would be protecting the buggy behavior

**Comment by abhijitxy** [comment]
Yeah @seveibar, I know. I'll fix the bug. Thanks for the catch!

File: tests/bug-builder/duplicate-port-hints.test.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  const soup = await pb
+    .add("bug", (cb) => {
+      cb.setProps({ footprint: "soic8" })
+    })
+    .build()
+
+  const sourcePorts = su(soup).source_port.list()
+
+  console.log("All source ports:", JSON.stringify(sourcePorts, null, 2))
```
Comment (seveibar) [inline tests/bug-builder/duplicate-port-hints.test.tsx]:
```suggestion
```

File: tests/bug-builder/duplicate-port-hints.test.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  sourcePorts.forEach((port) => {
+    console.log(`Port ${port.name} hints:`, port.port_hints)
```
Comment (seveibar) [inline tests/bug-builder/duplicate-port-hints.test.tsx]:
```suggestion
```

**Review by seveibar — CHANGES_REQUESTED** [review]

**Comment by seveibar** [comment]
Looks good, good fix. Normally i approve with suggestions but since this is an early pr i requested changes

**Review by seveibar — APPROVED** [review]
Lets make sure the tests pass but then looks good. I think they should autorun next time

**Comment by seveibar** [comment]
@abhijitxy i think you're failing format, formatbot usually fixes this but cant on forks iirc, try running nom run format
